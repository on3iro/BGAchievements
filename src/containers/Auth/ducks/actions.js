import * as types from './actionTypes'

export const loginSubmit = data => {
  return {
    type: types.LOGIN_SUBMIT,
    payload: data
  }
}

export const logout = () => {
  return {
    type: types.LOGOUT
  }
}

export const registerSubmit = data => {
  return {
    type: types.REGISTER_SUBMIT,
    payload: data
  }
}
