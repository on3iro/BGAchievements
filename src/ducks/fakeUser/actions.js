// @flow

import axios from 'axios'

import { API_URL } from 'config'

import type {
  Init,
  UserFetchFail,
  UserFetchSuccess
} from './types'

import { INIT, USER_FETCH_FAIL, USER_FETCH_SUCCESS } from './constants'

export const init = (): Init => {
  return {
    type: INIT
  }
}

export const fetchUser = (userId: string): Promise<*> => {
  return axios(`${API_URL}/${userId}`)
}

export const userFetchSuccessfulAction = (user: Object): UserFetchSuccess => {
  return {
    type: USER_FETCH_SUCCESS,
    payload: user.data
  }
}

export const userFetchFailedAction = (err: Object): UserFetchFail => {
  return {
    type: USER_FETCH_FAIL,
    payload: err
  }
}
