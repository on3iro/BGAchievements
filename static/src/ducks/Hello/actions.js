// @flow

import * as types from './types'

export const fetchSampleDataStart = (): types.FetchSampleDataStart => {
  return {
    type: 'SAMPLE_DATA_FETCH_START'
  }
}

export const fetchSampleDataSuccess = ({ data }: Object): types.FetchSampleDataSuccess => {
  return {
    type: 'SAMPLE_DATA_FETCH_SUCCESS',
    payload: data
  }
}

export const fetchSampleDataFailure = (error: Object) => {
  return {
    type: 'SAMPLE_DATA_FETCH_FAILURE',
    payload: error
  }
}
