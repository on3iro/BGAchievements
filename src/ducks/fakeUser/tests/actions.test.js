import * as actions from '../actions'

describe('FakeUser actionCreators', () => {
  it('Should create action to init fakeUser', () => {
    expect(actions.init())
      .toMatchSnapshot()
  })

  // TODO test for fetchUser (needs dependency injection)
  it('Should create action to fetch fakeUser')

  it('Should create action on fetch success', () => {
    expect(actions.userFetchSuccessfulAction({
      data: {
        name: 'User data...'
      }
    })).toMatchSnapshot()
  })

  it('Should create action on fetch fail', () => {
    expect(actions.userFetchFailedAction({ err: 'err' }))
      .toMatchSnapshot()
  })
}) // end actionCreators
