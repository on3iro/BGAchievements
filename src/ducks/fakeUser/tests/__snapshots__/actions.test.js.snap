// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FakeUser actionCreators Should create action on fetch fail 1`] = `
Object {
  "payload": Object {
    "err": "err",
  },
  "type": "USER_FETCH_FAIL",
}
`;

exports[`FakeUser actionCreators Should create action on fetch success 1`] = `
Object {
  "payload": Object {
    "name": "User data...",
  },
  "type": "USER_FETCH_SUCCESS",
}
`;

exports[`FakeUser actionCreators Should create action to init fakeUser 1`] = `
Object {
  "type": "INIT",
}
`;
